CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1929,6 +1934,18 @@ protected function processFormSubmission($contactID) {
     // fix currency ID
     $this->_params['currencyID'] = CRM_Core_Config::singleton()->defaultCurrency;
 
+    // CRM-18854
+    if (CRM_Utils_Array::value('adjust_recur_start_date', $this->_values)) {
+      $pledgeBlock = CRM_Pledge_BAO_PledgeBlock::getPledgeBlock($this->_id);
+      if (CRM_Utils_Array::value('start_date', $this->_params) || !CRM_Utils_Array::value('is_pledge_start_date_visible', $pledgeBlock)) {
+        $pledgeStartDate = CRM_Utils_Array::value('start_date', $this->_params, NULL);
+        $this->_params['receive_date'] = CRM_Pledge_BAO_Pledge::getPledgeStartDate($pledgeStartDate, $pledgeBlock);
+        $recurParams = CRM_Pledge_BAO_Pledge::buildRecurParams($this->_params);
+        $this->_params = array_merge($this->_params, $recurParams);
```

Comment:
Yes, exactly right. The problem occurred in the payment processor class where the code needed contributionRecurID to be present in the params. Specifically, this line: https://github.com/civicrm/civicrm-core/blob/4.7.9-rc/CRM/Core/Payment/AuthorizeNet.php#L121

I will follow the steps above to attempt to refactor the code.